packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -443,24 +456,52 @@ export class Nodegroup extends Resource implements INodegroup {
-    INSTANCE_TYPES.graviton.includes(instanceType.toString().substring(0, 2)) ? NodegroupAmiType.AL2_ARM_64 :
-      INSTANCE_TYPES.gpu.includes(instanceType.toString().substring(0, 2)) ? NodegroupAmiType.AL2_X86_64_GPU :
-        INSTANCE_TYPES.inferentia.includes(instanceType.toString().substring(0, 4)) ? NodegroupAmiType.AL2_X86_64_GPU :
-          NodegroupAmiType.AL2_X86_64;
+/**
+ * AMI types of different architectures. Make sure AL2 is always the first element, which will be the default AmiType if amiType and
+ * launchTemplateSpec are both undefined.
+ */
+const arm64AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_ARM_64, NodegroupAmiType.BOTTLEROCKET_ARM_64];
+const x8664AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_X86_64, NodegroupAmiType.BOTTLEROCKET_X86_64];
+const gpuAmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_X86_64_GPU];
+
+/**
+ * This function returns an array of "possible AMI types" from the given instance type
+ * @param instanceType The instance type
+ * @returns NodegroupAmiType[]
+ */
+function getPossibleAmiTypesForInstanceType(instanceType: InstanceType): NodegroupAmiType[] {
+  return INSTANCE_TYPES.graviton2.includes(instanceType.toString().substring(0, 3)) ? arm64AmiTypes :
+    INSTANCE_TYPES.graviton.includes(instanceType.toString().substring(0, 2)) ? arm64AmiTypes :
+      INSTANCE_TYPES.gpu.includes(instanceType.toString().substring(0, 2)) ? gpuAmiTypes :
+        INSTANCE_TYPES.inferentia.includes(instanceType.toString().substring(0, 4)) ? gpuAmiTypes :
+          x8664AmiTypes;
 }
 
-// this function examines the CPU architecture of every instance type and determines
-// what ami type is compatible for all of them. it either throws or produces a single value because
-// instance types of different CPU architectures are not supported.
-function getAmiType(instanceTypes: InstanceType[]) {
-  const amiTypes = new Set(instanceTypes.map(i => getAmiTypeForInstanceType(i)));
-  if (amiTypes.size == 0) { // protective code, the current implementation will never result in this.
+/**
+ * This function examines the CPU architecture of every instance type and determines
+ * what AMI types are compatible for all of them. it either throws or produces an array of possible AMI types because
+ * instance types of different CPU architectures are not supported.
+ * @param instanceTypes The instance types
+ * @returns NodegroupAmiType[]
+ */
+function getPossibleAmiTypes(instanceTypes: InstanceType[]): NodegroupAmiType[] {
+  const amiTypes = new Set<NodegroupAmiType>();
+  for (const t of instanceTypes) {
+    getPossibleAmiTypesForInstanceType(t).forEach(x => amiTypes.add(x));
+  }
+  if (amiTypes.size === 0) { // protective code, the current implementation will never result in this.
     throw new Error(`Cannot determine any ami type comptaible with instance types: ${instanceTypes.map(i => i.toString).join(',')}`);
   }
-  if (amiTypes.size > 1) {
+  let cpuArchTypes: number = 0;
```

Comment:
Let me suggest a slightly different way, that IMO is easier to understand:
1. From the list of instance types, compute the set of architectures (see my other comment).
2. If the set has 0 or more than one element, throw an error.
3. At this point, there is only one architecture. Return the list of types that correspond to it (combine `arm64AmiTypes`, `x8664AmiTypes` and `gpuAmiTypes` into a single `Map<InstanceArchitecture, NodegroupAmiType[]>`, for example).

Reply:
fixed